### PR TITLE
Typing the app

### PR DESCRIPTION
**Pull request by jrodva**
#### What does this PR do?
This pr includes and use new types in the project, the scope has been avoid _**any type**_ in the controller layer, however there are some changes in the service layer. Last but not least, we have a new rule in the tslint file to avoid new _**any type**_ in the code.

#### Where should the reviewer start?
- src/app/definitions/**

#### What is missing?
I had to reduce the scope because the task was coming infinite, so there are some // tslint:disable:no-any in the project, we'll delete gradually them. 

#### How should this be manually tested?
1. npm start
2. Navigate across the app and verify the most important operations related to deployment, graph painting, labels interaction, instance view info, registered info, user management, devices management, etc
 
#### What are the relevant tickets?
- [NP-2067](https://nalej.atlassian.net/browse/NP-2067)

### CODE REVIEW

File: src/app/infrastructure/infrastructure.service.ts [hunk 1]
```diff
@@ -42,7 +43,7 @@ export class InfrastructureService {
    * @param total Number of total ECs online
    * @returns anonym array with the required object structure for pie chart rendering
    */
-  generateSummaryChartData(online: number, total: number): any[] {
+  generateSummaryChartData(online: number, total: number): ChartData[] {
```
[the remark this answers is not in the document]
Comment (jrodva) [reply src/app/infrastructure/infrastructure.service.ts]: Done :D

File: src/app/applications/applications.component.ts [hunk 2]
```diff
@@ -816,7 +808,7 @@ export class ApplicationsComponent extends ToolsComponent implements OnInit, OnD
    * @param instance Instance to generate the node instance
    * @param cluster Cluster for relate with our node instance
    */
-  private addNodeInstance(instance, cluster): any {
+  private addNodeInstance(instance: ApplicationInstance, cluster: Cluster): GraphNode & StyledNode {
```
[the remark this answers is not in the document]
Comment (jrodva) [reply src/app/applications/applications.component.ts]: Done !!!

File: src/app/applications/applications.component.ts [hunk 3]
```diff
@@ -15,7 +15,6 @@ import { Component, OnInit, OnDestroy } from '@angular/core';
 import { BackendService } from '../services/backend.service';
```
[the remark this answers is not in the document]
Comment (jrodva) [reply src/app/applications/applications.component.ts]: I can not reproduce it, please give me more info.

File: src/app/applications/deploy-instance/deploy-instance.component.ts [hunk 4]
```diff
@@ -413,7 +419,7 @@ export class DeployInstanceComponent implements OnInit {
   isInactiveNext(f) {
     let isInactiveNext = false;
     if (this.conditionExpression === 'basic') {
-      isInactiveNext = !this.selectedApp || this.selectedApp.id === -1 || !f.instanceName.value;
+      isInactiveNext = !this.selectedApp || this.selectedApp.getId() === '-1' || !f.instanceName.value;
```
[the remark this answers is not in the document]
Comment (jrodva) [reply src/app/applications/deploy-instance/deploy-instance.component.ts]: Done :D

File: src/app/applications/app-info/app-info.component.ts [hunk 5]
```diff
@@ -34,8 +38,8 @@ const NOTIFICATION_TIMEOUT = 3000; // 3 seconds
 })
 
 export class AppInfoComponent implements OnChanges {
-  @Input() instance: any;
-  @Input() registeredData: any;
+  @Input() instance: ApplicationInstance;
```
Comment (bzapico) [inline src/app/applications/app-info/app-info.component.ts]: Error when trying to navigate to a specific instance info view:
Captura de pantalla 2019-12-02 a las 17 09 27

Comment (bzapico) [reply src/app/applications/app-info/app-info.component.ts]: This was an error in an environment with errors

File: src/app/applications/app-info/app-info.component.ts [hunk 6]
```diff
@@ -34,8 +38,8 @@ const NOTIFICATION_TIMEOUT = 3000; // 3 seconds
...
+  @Input() registeredData: ApplicationDescriptor;
```
Comment (bzapico) [inline src/app/applications/app-info/app-info.component.ts]: Disconnect in app-info component throws an error:
Captura de pantalla 2019-12-02 a las 17 36 50

Comment (jrodva) [reply src/app/applications/app-info/app-info.component.ts]: Solved !!!

**Comment by jrodva** [comment]
> Captura de pantalla 2019-12-04 a las 15 15 24
> Error un app-info.component.ts when clicked in instance view to redirect to its descriptor view.

Fixed !!!

**Comment by bzapico** [comment]
LGTM

**Pull request merged**